core/src/main/java/com/expediagroup/streamplatform/streamregistry/core/services/StreamService.java

```diff
@@ -93,7 +102,17 @@
 
   @PreAuthorize("hasPermission(#stream, 'DELETE')")
   public void delete(Stream stream) {
-    throw new UnsupportedOperationException();
+    handlerService.handleDelete(stream);
+    Schema schema = new Schema(stream.getSchemaKey(), null,null);
+    consumerRepository.findAllAndDelete(stream.getKey());
+    producerRepository.findAllAndDelete(stream.getKey());
+    streamBindingRepository.findAllAndDelete(stream.getKey());
+    streamRepository.delete(stream);
+    List<Stream> streams = streamRepository.findAll().stream().filter(s -> s.getSchemaKey()
+      .equals(stream.getSchemaKey())).collect(Collectors.toList());
+    if(streams.isEmpty()) {
+      schemaService.delete(schema);
```

Comment:
I'm not sure the `isEmpty` check is enough here. Due to the async nature of the delete function (it creates a collection of futures but never waits for them to be completed) it's possible (though maybe not likely) that those deletes haven't been published to Kafka or Consumed from Kafka. 

I think Elliots suggestion above (should only be deleted if `this` or 0 Streams reference the schema) solves this. We still trigger the delete if the stream is the only thing referencing the schema

Reply:
added check for either, has current stream being deleted in the list or its empty